### PR TITLE
#168978572 Create user account

### PR DESCRIPTION
**Pull request by hirwaf**
#### What does this PR do?
Having created an account API endpoint working `/api/v2/auth/signup/`
#### Description of Task to be completed?
- change user controller
- add util for inserting data
- add util for finding By column
- add eslint rules
- change helper into class
#### How should this be manually tested?
- Using Postman test above endpoint with this header
key: **Content-Type** 
value: **application/json**
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#168978572
#### Screenshots (if appropriate)
- On Success
Screenshot_20191008_140810_email_saved
- When Email exists
Screenshot_20191008_140532_when_email_exists
#### Questions:
N/A

### CODE REVIEW

File: server/controllers/UserController.js [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
+    const checkEmail = await user.getByEmail(data.email);
+    Helpers.dbError(response, checkEmail);
+    if (checkEmail.count > 0) return Helpers.sendResponse(response, 419, 'Email already exists !');
+    const saveUser = await user.create(data);
+    if (saveUser.errors) return Helpers.dbError(response, saveUser);
+    const token = Helpers.generateToken(saveUser.rows[0].id);
     return Helpers.sendResponse(response, 201, 'User created successfully', { token });
   }
 
   static signIn(request, response) {
```
Comment (ekumamait) [inline server/controllers/UserController.js:20]: This Pull Request is for `Sign Up` so am wondering why we have `Sign In` functionality here?

File: server/helpers/Helpers.js [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
+  }
+
+  static comparePassword(hashPassword, password) {
     return bcrypt.compareSync(password, hashPassword);
-  },
-  generateToken(userId) {
+  }
+
+  static generateToken(userId) {
     // const random = Math.random().toString(36).substring(7);
```
Comment (ekumamait) [inline server/helpers/Helpers.js:14]: why do you have commented code?

**Review by ekumamait — CHANGES_REQUESTED** [review]
Add screenshots showing the payload you're sending in `Postman` and the response and also the storyID show be a link to that Pivotal tracker story itself

File: server/controllers/UserController.js (hunk 1 above)
Comment (hirwaf) [reply server/controllers/UserController.js:20]: it was for the previous challenge

File: server/helpers/Helpers.js (hunk 2 above)
Comment (hirwaf) [reply server/helpers/Helpers.js:14]: I'm going to remove them ASAP

**Comment by ekumamait** [comment]
I think the right status code to use for instances where the email already exists would be conflict `409`. However, why did you prefer to use `419`

**Pull request merged**